gui/rendering/Node.cpp

```diff
@@ -0,0 +1,160 @@
+//
+// Created by hcaramizaru on 4/15/24.
+//
+
+#include "Node.hpp"
+#include "sme/logger.hpp"
+
+namespace rendering {
+
+Node::Node(const std::string &name, const QVector3D &position,
+           const QVector3D &rotation, const QVector3D &scale)
+    : name(name), m_position(position), m_rotation(rotation), m_scale(scale) {
+
+  this->worldTransform.setToIdentity();
+  this->localTransform.setToIdentity();
+}
+
+Node::Node(const std::string &name, const QVector3D &position,
+           const QVector3D &rotation)
+    : Node(name, position, rotation, QVector3D(1, 1, 1)) {}
+
+Node::Node(const std::string &name, const QVector3D &position)
+    : Node(name, position, QVector3D(0, 0, 0)) {}
+
+Node::Node(const std::string &name) : Node(name, QVector3D(0, 0, 0)) {}
+
+Node::Node() : Node("Node") {}
```

Comment:
It would be. I will keep it like this, for the moment, as there is a good chance to change it once the scene graph is integrated into the OpenGL widget.